packages/store/src/scalarMap.js

```diff
@@ -0,0 +1,89 @@
+// Copyright (C) 2019 Agoric, under Apache license 2.0
+
+// @ts-check
+
+import { assert, details as X, q } from '@agoric/assert';
+import { passStyleOf } from '@agoric/marshal';
+import { mustBeComparable } from '../../same-structure';
+
+const assertKey = key => {
+  harden(key); // TODO: Just a transition kludge. Remove when possible.
+  mustBeComparable(key);
+  const passStyle = passStyleOf(key);
+  switch (passStyle) {
+    case 'bigint':
+    case 'boolean':
+    case 'null':
+    case 'number':
+    case 'string':
+    case 'symbol':
+    case 'undefined':
+    case 'remotable': {
+      return;
+    }
+    case 'copyArray':
+    case 'copyRecord':
+    case 'copyError': {
+      assert.fail(X`composite keys not yet allowed: ${key}`);
+    }
+    // case 'promise': is precluded by `mustBeComparable` above
+    default: {
+      assert.fail(X`unexpected passStyle ${passStyle}`);
+    }
+  }
+};
+
+const assertValue = value => {
+  harden(value); // TODO: Just a transition kludge. Remove when possible.
+  passStyleOf(value); // asserts that value is passable
+};
+
+/**
+ * Distinguishes between adding a new key (init) and updating or
+ * referencing a key (get, set, delete).
+ *
+ * `init` is only allowed if the key does not already exist. `Get`,
+ * `set` and `delete` are only allowed if the key does already exist.
+ *
+ * @template K,V
+ * @param {string} [keyName='key'] - the column name for the key
+ * @param {Partial<StoreOptions>=} _options
+ * @returns {Store<K,V>}
+ */
+export const makeScalarMap = (keyName = 'key', _options = {}) => {
```

Comment:
Can you explain what is "scalar" about it?

Reply:
Done
```js
 * This is a *scalar* map in that the keys can only be atomic values, primitives
 * or remotables. Other storeMaps will accept, for example, copyArrays and
 * copyRecords, as keys and look them up based on equality of their contents.
```
Resolving.